dali/operators/util/normal_distribution_op.h

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_UTIL_NORMAL_DISTRIBUTION_OP_H_
+#define DALI_OPERATORS_UTIL_NORMAL_DISTRIBUTION_OP_H_
+
+#include <random>
+#include <vector>
+#include "dali/core/convert.h"
+#include "dali/pipeline/operator/operator.h"
+
+namespace dali {
+namespace detail {
+
+/**
+ * Names of arguments
+ */
+const std::string kMean = "mean";      // NOLINT
+const std::string kStddev = "stddev";  // NOLINT
+const std::string kDtype = "dtype";    // NOLINT
+const int kNumOutputs = 1;
+
+}  // namespace detail
+
+template<typename Backend>
+class NormalDistribution : public Operator<Backend> {
+ public:
+  ~NormalDistribution() override = default;
+
+  DISABLE_COPY_MOVE_ASSIGN(NormalDistribution);
+
+ protected:
+  explicit NormalDistribution(const OpSpec &spec) :
+          Operator<Backend>(spec),
+          mean_(spec.GetArgument<std::remove_const_t<decltype(this->mean_)>>(detail::kMean)),
+          stddev_(spec.GetArgument<std::remove_const_t<decltype(this->stddev_)>>(detail::kStddev)),
+          seed_(spec.GetArgument<std::remove_const_t<decltype(this->seed_)>>("seed")),
+          dtype_(spec.GetArgument<std::remove_const_t<decltype(this->dtype_)>>(detail::kDtype)) {}
+
+
+  bool CanInferOutputs() const override {
+    return true;
+  }
+
+
+  USE_OPERATOR_MEMBERS();
+  const float mean_, stddev_;
+  const int64_t seed_;
+  const DALIDataType dtype_;
+};
+
+
+class NormalDistributionCpu : public NormalDistribution<CPUBackend> {
+ public:
+  explicit NormalDistributionCpu(const OpSpec &spec) : NormalDistribution(spec), rng_(),
```

Comment:
Please use the seed argument to initialize the rng. Seed is available in all operators.

Reply:
Will do

Reply:
Done